pkg/dart/lock/parse.go

```diff
@@ -0,0 +1,82 @@
+package lock
+
+import (
+	"fmt"
+	dio "github.com/aquasecurity/go-dep-parser/pkg/io"
+	"github.com/aquasecurity/go-dep-parser/pkg/types"
+	"golang.org/x/xerrors"
+	"gopkg.in/yaml.v3"
+	"strings"
+)
+
+const (
+	idFormat      = "%s@%s"
+	directMainDep = "direct main"
+	directDevDep  = "direct dev"
+)
+
+type Parser struct{}
+
+func NewParser() types.Parser {
+	return &Parser{}
+}
+
+type lock struct {
+	Packages map[string]Dep `yaml:"packages"`
+}
+
+type Dep struct {
+	Dependency string `yaml:"dependency"`
+	Version    string `yaml:"version"`
+}
+
+func (Parser) Parse(r dio.ReadSeekerAt) ([]types.Library, []types.Dependency, error) {
+	l := &lock{}
+	decoder := yaml.NewDecoder(r)
+	if err := decoder.Decode(&l); err != nil {
+		return nil, nil, xerrors.Errorf("failed to decode pubspec.lock file: %s", err.Error())
+	}
+	var libs []types.Library
+	for name, dep := range l.Packages {
+		lib := types.Library{
+			ID:      fmt.Sprintf(idFormat, name, dep.Version),
+			Name:    name,
+			Version: dep.Version,
+			// there are 3 types of dependency fields:
+			// "direct main", "direct dev" and transitive
+			// transitive non-string field
```

Comment:
What do you mean? Isn't it string?
https://github.com/aquasecurity/go-dep-parser/blob/b4aa3d0aeaba9085595c32c48fdd3ae1d5ca3584/pkg/dart/lock/testdata/happy.lock#L17